backend/assignments/1/Assignment-1-kdu/src/main/java/org/example/Transactions/AddVolumeTransaction.java

```diff
@@ -0,0 +1,39 @@
+package org.example.Transactions;
+
+import org.example.delay.HashFun;
+import org.example.entities.AddVolume;
+import org.example.entities.UpdatePrice;
+
+public class AddVolumeTransaction extends Thread {
+    private AddVolume addVolume;
+    private String transactionID;
+
+    public AddVolume getUpdatePrice() {
+        return addVolume;
+    }
+
+    public AddVolumeTransaction(AddVolume addVolume) {
+        this.addVolume = addVolume;
+    }
+
+    @Override
+    public void run() {
+        synchronized (this.addVolume.getCoin()) {
```

Comment:
What if user1 needs to work on Coin1 and user2 on coin2.
But user2 will still be in wait state untill the method reopoens lock.
Can the locking be in a better place?